File: src/main/java/io/sailex/aiNpc/model/NPC.java
package io.sailex.aiNpc.model;

import io.sailex.aiNpc.llm.ILLMService;
import io.sailex.aiNpc.npc.NPCContextGenerator;
import io.sailex.aiNpc.npc.NPCController;
import io.sailex.aiNpc.npc.NPCEntity;
import java.util.UUID;
import lombok.AllArgsConstructor;
import lombok.Data;

@Data
@AllArgsConstructor
public class NPC {

	private UUID id;
	private NPCEntity npcEntity;
	private NPCController npcController;
	private ILLMService llmService;
	private NPCContextGenerator npcContextGenerator;
}

File: ai-npc-client/src/main/java/io/sailex/ai/npc/client/model/NPC.kt
package io.sailex.ai.npc.client.model

import io.sailex.ai.npc.client.context.ContextGenerator
import io.sailex.ai.npc.client.llm.ILLMClient
import io.sailex.ai.npc.client.npc.NPCController
import net.minecraft.client.network.ClientPlayerEntity
import java.util.UUID

/**
 * Represents an NPC in the game (client player entity) and its llm service, context generator, and controller
 */
class NPC(
    val id: UUID,
    val npcEntity: ClientPlayerEntity,
    val npcController: NPCController,
    val contextGenerator: ContextGenerator,
    val llmService: ILLMClient,
)
